src/bundle/DependencyInjection/Compiler/AuthenticationProviderDecoratorCompilerPass.php

```diff
@@ -17,6 +17,12 @@ class AuthenticationProviderDecoratorCompilerPass implements CompilerPassInterfa
 {
     public function process(ContainerBuilder $container): void
     {
+        if ($container->hasDefinition('security.authenticator.manager')) {
+            // Skip integration with the old security system when the new authenticator manager is present
+            // Any better way to do this?
+            return;
+        }
+
```

Comment:
What's the best way for me as as 3rd party bundle to find out if the authenticator or the old system is used? I don't have access for the security bundle configuration, or do I?

For compliler passes I can check for `hasDefinition('security.authenticator.manager')`, since they're applied after the security bundle was loaded.

Unfortunately this doesn't work in
- `DependencyInjection/Factory/Security/TwoFactorFactory.php`
- `DependencyInjection\SchebTwoFactorExtension.php`
- `SchebTwoFactorBundle.php`

because when the bundle loads, the service definitions from the security bundle haven't necessary loaded yet.

Reply:
Checking for the `security.authenticator.manager` service definition in a compiler pass is imho the correct solution. For the other cases:

* The `createAuthenticator()` method is only called when using the authenticator and the `create()` method is only called when using the old authenticator. So there is no need for checking it in these methods.
* I don't see you using it in the extension atm. If you need it, I would say: populate the container with both the old and new system. As the services are private, they will be removed when not used (and otherwise, you can use a compiler pass to remove/tweak the definitions).
* The bundle should not need to know it as well, the compiler pass can check for the service ID and return early if the old system is used.

Reply:
> I don't see you using it in the extension atm. If you need it, I would say: populate the container with both the old and new system. As the services are private, they will be removed when not used (and otherwise, you can use a compiler pass to remove/tweak the definitions).

I thought I can skip some unnecessary service definitions ;). But I'm also fine with just loading them, even when they're not used.